frontend/src/components/useProducts.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react'
+import { getAllProducts, getProductsByUserId } from '../communications/userApi'
+
+export const GetProducts = () => {
+  const [listOfAllProducts, setListOfAllProducts] = useState([])
+
+  const fetchProducts = async () => {
+    try {
+      const fetchedProducts = await getAllProducts()
+      setListOfAllProducts(fetchedProducts)
+    } catch (error) {
+      throw new Error('Failed to get products.')
+    }
+  }
+
+  useEffect(() => {
+    fetchProducts()
+  }, [])
+
+  return { listOfAllProducts }
+}
+
+export const GetProductsById = () => {
```

Comment:
Ezt is use-al nevezzük el: useProductById

Illetve ha by ID, akkor az egyte ad vissza, ne legyen többesszám.

Reply:
Ááá, most értettem meg, hogy nem product ID alapján, hanem company ID alapján. Akkor meg legyen ez a neve: useProductsForCompany.